contracts/src/assertionStakingPool/AssertionStakingPool.sol

```diff
@@ -105,32 +67,38 @@ contract AssertionStakingPool {
 
         poolState = PoolState.ASSERTED;
         // approve spending from rollup for newStakeOnNewAssertion call
-        stakeToken.approve(rollup, requiredStake);
+        stakeToken.safeIncreaseAllowance(rollup, requiredStake);
         IRollupUser(rollup).newStakeOnNewAssertion(requiredStake, assertionInputs, assertionHash);
         emit AssertionCreated();
     }
 
-    /// @notice Move stake back from rollup contract to this contract. Calalble only if this contract has already created an assertion and it's been confirmed.
-    function returnOldStakeBackToPool() external {
+    /// @notice update pool state if assertion is no longer active (confirmed or has a child) , and make deposit withdrawable.
+    function setPoolStateInactive() external {
         if (poolState != PoolState.ASSERTED) {
             revert PoolNotInAssertedState(poolState);
-        }
+        }   
 
-        poolState = PoolState.CONFIRMED;
+        poolState = PoolState.INACTIVE;
 
-        if (IRollupCore(rollup).getAssertion(assertionHash).status != AssertionStatus.Confirmed) {
-            revert AssertionNotConfirmed(assertionHash);
+        if (!IRollupCore(rollup).stakerIsInactive(address(this))) {
+            revert AssertionNotInactive(assertionHash);
         }
         IRollupUser(rollup).returnOldDeposit();
+        emit PoolStateInactive();
+    }
+
+    /// @notice Move stake back from rollup contract to this contract. Calalble only if this contract has already created an assertion and it's now inactive.
+    /// @dev Separate call from setPoolStateInactive since withdrawStakerFunds reverts with 0 balance (in e.g., case of admin forceRefundStaker)
+    function returnOldStakeBackToPool() external {
+        if (poolState != PoolState.INACTIVE) {
+            revert PoolNotInInactiveState(poolState);
+        }
```

Comment:
should remove this check or call `withdrawStakerFunds` after `returnOldDeposit` in `setPoolStateInactive`

Reply:
bahh yeah ok i don't think we even need the "INACTIVE" poolstate

Reply:
....in fact, can just get rid of poolstate altogether, much simpler